rocAL/source/readers/file_source_reader.cpp

```diff
@@ -78,6 +70,17 @@ Reader::Status FileSourceReader::initialize(ReaderConfig desc) {
 void FileSourceReader::incremenet_read_ptr() {
     _read_counter++;
     _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();
+    if (_last_batch_info.first == RocalBatchPolicy::DROP) {
+        if (_last_batch_info.second == true) {
+            if ((_file_names.size() / _batch_count) == _curr_file_idx)  // To check if it the last batch
```

Comment:
Not very clear what this is doing

Reply:
Updated the comments for better understanding